jupyterhub/user.py

```diff
@@ -235,8 +235,11 @@ def spawn(self, options=None):
         api_token = self.new_api_token()
         db.commit()
 
+        self.server_base_url = base_url
```

Comment:
User.server_base_url probably shouldn't be defined. With multiple servers, this no longer makes sense.